utils/zerovec/src/ule/macros.rs

```diff
@@ -9,9 +9,31 @@
 /// Pass any (cheap to construct) value.
 #[macro_export]
 macro_rules! impl_ule_from_array {
+    ($aligned:ty, $unaligned:ty, $default:expr, $single:path, $fn_name:ident) => {
+        #[doc = concat!("Convert an array of `", stringify!($aligned), "` to an array of `", stringify!($unaligned), "`.")]
+        pub const fn $fn_name<const N: usize>(arr: [$aligned; N]) -> [Self; N] {
+            let mut result = [$default; N];
```

Comment:
thought: could we potentially use `MaybeUninit` to avoid the need for `$default`? arrays of `MaybeUninit` are transmutable to regular arrays